tests/integration_tests/performance/test_block_ab.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Performance benchmark for block device emulation."""
+
+import concurrent
+import os
+import shutil
+from pathlib import Path
+
+import pytest
+
+import host_tools.drive as drive_tools
+from framework.properties import global_props
+from framework.utils import CmdBuilder, get_cpu_percent, run_cmd
+
+# size of the block device used in the test, in MB
+BLOCK_DEVICE_SIZE_MB = 2048
+
+# Time (in seconds) for which fio "warms up"
+WARMUP_SEC = 10
+
+# Time (in seconds) for which fio runs after warmup is done
+RUNTIME_SEC = 30
+
+# VM guest memory size
+GUEST_MEM_MIB = 1024
+
+
+def prepare_microvm_for_test(microvm):
+    """Prepares the microvm for running a fio-based performance test by tweaking
+    various performance related parameters."""
+    rc, _, stderr = microvm.ssh.run("echo 'none' > /sys/block/vdb/queue/scheduler")
+    assert rc == 0, stderr
+    assert stderr == ""
+
+    # First, flush all guest cached data to host, then drop guest FS caches.
+    rc, _, stderr = microvm.ssh.run("sync")
+    assert rc == 0, stderr
+    assert stderr == ""
+    rc, _, stderr = microvm.ssh.run("echo 3 > /proc/sys/vm/drop_caches")
+    assert rc == 0, stderr
+    assert stderr == ""
+
+    # Then, flush all host cached data to hardware, also drop host FS caches.
+    run_cmd("sync")
+    run_cmd("echo 3 > /proc/sys/vm/drop_caches")
+
+
+def run_fio(microvm, mode, block_size):
+    """Run a fio test in the specified mode with block size bs."""
+    cmd = (
+        CmdBuilder("fio")
+        .with_arg(f"--name={mode}-{block_size}")
+        .with_arg(f"--rw={mode}")
+        .with_arg(f"--bs={block_size}")
+        .with_arg("--filename=/dev/vdb")
+        .with_arg("--time_base=1")
+        .with_arg(f"--size={BLOCK_DEVICE_SIZE_MB}M")
+        .with_arg("--direct=1")
+        .with_arg("--ioengine=libaio")
+        .with_arg("--iodepth=32")
+        .with_arg(f"--ramp_time={WARMUP_SEC}")
+        .with_arg(f"--numjobs={microvm.vcpus_count}")
+        # Set affinity of the entire fio process to a set of vCPUs equal in size to number of workers
+        .with_arg(
+            f"--cpus_allowed={','.join(str(i) for i in range(microvm.vcpus_count))}"
+        )
+        # Instruct fio to pin one worker per vcpu
+        .with_arg("--cpus_allowed_policy=split")
+        .with_arg("--randrepeat=0")
+        .with_arg(f"--runtime={RUNTIME_SEC}")
+        .with_arg(f"--write_bw_log={mode}")
+        .with_arg("--log_avg_msec=1000")
+        .with_arg("--output-format=json+")
+        .build()
+    )
+
+    logs_path = Path(microvm.jailer.chroot_base_with_id()) / "fio_output"
```

Comment:
Could you add a comment to why the construction of this is changed from https://github.com/firecracker-microvm/firecracker/blob/main/tests/integration_tests/performance/test_block_performance.py#L78C71-L78C73